string source_color
string destination_color

